src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -0,0 +1,342 @@
+import logging
+import time
+from datetime import datetime, timedelta
+from typing import Any, TypedDict
+
+import sentry_sdk
+from django.conf import settings
+from google.api_core.exceptions import DeadlineExceeded, ServiceUnavailable
+from snuba_sdk import Column, Condition, Entity, Function, Op, Query, Request
+from snuba_sdk.orderby import Direction, OrderBy
+
+from sentry import features, nodestore
+from sentry.api.endpoints.group_similar_issues_embeddings import get_stacktrace_string
+from sentry.eventstore.models import Event
+from sentry.grouping.grouping_info import get_grouping_info
+from sentry.issues.grouptype import ErrorGroupType
+from sentry.issues.occurrence_consumer import EventLookupError
+from sentry.models.group import Group
+from sentry.models.grouphash import GroupHash
+from sentry.models.project import Project
+from sentry.seer.utils import (
+    CreateGroupingRecordData,
+    CreateGroupingRecordsRequest,
+    post_bulk_grouping_records,
+)
+from sentry.silo.base import SiloMode
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+from sentry.tasks.base import instrumented_task
+from sentry.utils import json, metrics, redis
+from sentry.utils.iterators import chunked
+from sentry.utils.query import RangeQuerySetWrapper
+from sentry.utils.safe import get_path
+from sentry.utils.snuba import bulk_snuba_queries
+
+ITERATOR_CHUNK = 20
+SEER_BACKFILL_DELAY_PER_RECORD = 0.1
+BACKFILL_NAME = "backfill_grouping_records"
+LAST_PROCESSED_REDIS_KEY = "grouping_record_backfill.last_processed_id"
+
+logger = logging.getLogger(__name__)
+
+
+class GroupEventRow(TypedDict):
+    event_id: str
+    group_id: int
+    message: str
+
+
+class GroupStacktraceData(TypedDict):
+    data: list[CreateGroupingRecordData]
+    stacktrace_list: list[str]
+
+
+@instrumented_task(
+    name="sentry.tasks.backfill_seer_grouping_records",
+    queue="default",
+    max_retries=0,
+    silo_mode=SiloMode.REGION,
+)
+@metrics.wraps(f"{BACKFILL_NAME}.task")
+def backfill_seer_grouping_records(project: Project, *args: Any, **kwargs: Any) -> None:
+    if not features.has("projects:similarity-embeddings-grouping", project):
+        return
+
+    redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+    last_processed_id = int(redis_client.get(LAST_PROCESSED_REDIS_KEY) or 0)
+
+    for group_id_message_data_batch in chunked(
+        RangeQuerySetWrapper(
+            Group.objects.filter(
+                project_id=project.id, id__gt=last_processed_id, type=ErrorGroupType.type_id
+            ).values_list("id", "message", "data"),
+            result_value_getter=lambda item: item[0],
+        ),
+        ITERATOR_CHUNK,
+    ):
+        backfill_chunked_seer_grouping_records.apply_async(
+            args=[group_id_message_data_batch, project.id],
+            countdown=ITERATOR_CHUNK * SEER_BACKFILL_DELAY_PER_RECORD,
+        )
```

Comment:
This way will work ok, although one concern I have is that with a lot of schedule tasks pending we might take up a lot of memory in rabbit/celery.  Here's another option

```
def backfill_seer_grouping_records(project: Project, last_processed_id=None, *args: Any, **kwargs: Any) -> None:
    ...
    if last_processed_id is None:
        last_processed_id = int(redis_client.get(LAST_PROCESSED_REDIS_KEY) or 0)

    MAX_CHUNKS = X
    for i, chunk in enumerate(chunked):
        # Call seer function
        last_processed_id = group_id_message_data_batch[-1][0]
        redis_client.set(f"{LAST_PROCESSED_REDIS_KEY}", last_processed_id, ex=60 * 60 * 24 * 7)
        if i >= MAX_CHUNKS:
            backfill_seer_grouping_records.delay(project_id, last_processed_id=last_processed_id)
            return
```

Since we don't actually want to run these in parallel, it might be better to handle it this way.